crates/lox_core/src/bodies.rs

```diff
@@ -62,8 +64,223 @@ pub fn gravitational_parameter<T: PointMass>(_: T) -> f64 {
+        let theta = Self::theta(t);
+        let mut c_trig = [0.0; N_COEFFICIENTS];
+        for i in 0..N_COEFFICIENTS {
+            c_trig[i] = c[i] * theta[i].sin();
+        }
+        let c_trig: f64 = c_trig.iter().sum();
+        c0 + c1 * t / dt + c2 * t.powi(2) / dt.powi(2) + c_trig
+    }
+
+    fn right_ascension_dot(t: f64) -> f64 {
+        let dt = SECONDS_PER_JULIAN_CENTURY;
+        let (_, c1, c2, c) = Self::right_ascension_coefficients();
+        let (_, theta1) = Self::nutation_precession_coefficients();
+        let theta = Self::theta(t);
+        let mut c_trig = [0.0; N_COEFFICIENTS];
+        for i in 0..N_COEFFICIENTS {
+            c_trig[i] = c[i] * theta1[i] / SECONDS_PER_JULIAN_CENTURY * theta[i].cos()
+        }
+        let c_trig: f64 = c_trig.iter().sum();
+        c1 / dt + 2.0 * c2 * t / dt.powi(2) + c_trig
+    }
+
+    fn declination(t: f64) -> f64 {
+        let dt = SECONDS_PER_JULIAN_CENTURY;
+        let (c0, c1, c2, c) = Self::declination_coefficients();
+        let theta = Self::theta(t);
+        let mut c_trig = [0.0; N_COEFFICIENTS];
+        for i in 0..N_COEFFICIENTS {
+            c_trig[i] = c[i] * theta[i].cos();
+        }
+        let c_trig: f64 = c_trig.iter().sum();
+        c0 + c1 * t / dt + c2 * t.powi(2) / dt.powi(2) + c_trig
+    }
+
+    fn declination_dot(t: f64) -> f64 {
+        let dt = SECONDS_PER_JULIAN_CENTURY;
+        let (_, c1, c2, c) = Self::declination_coefficients();
+        let (_, theta1) = Self::nutation_precession_coefficients();
+        let theta = Self::theta(t);
+        let mut c_trig = [0.0; N_COEFFICIENTS];
+        for i in 0..N_COEFFICIENTS {
+            c_trig[i] = c[i] * theta1[i] / SECONDS_PER_JULIAN_CENTURY * theta[i].sin()
+        }
+        let c_trig: f64 = c_trig.iter().sum();
+        c1 / dt + 2.0 * c2 * t / dt.powi(2) - c_trig
+    }
+
+    fn prime_meridian(t: f64) -> f64 {
+        let dt = SECONDS_PER_DAY;
+        let (c0, c1, c2, c) = Self::prime_meridian_coefficients();
+        let theta = Self::theta(t);
+        let mut c_trig = [0.0; N_COEFFICIENTS];
+        for i in 0..N_COEFFICIENTS {
+            c_trig[i] = c[i] * theta[i].sin();
+        }
+        let c_trig: f64 = c_trig.iter().sum();
+        c0 + c1 * t / dt + c2 * t.powi(2) / dt.powi(2) + c_trig
+    }
+
+    fn prime_meridian_dot(t: f64) -> f64 {
+        let dt = SECONDS_PER_DAY;
+        let (_, c1, c2, c) = Self::prime_meridian_coefficients();
+        let (_, theta1) = Self::nutation_precession_coefficients();
+        let theta = Self::theta(t);
+        let mut c_trig = [0.0; N_COEFFICIENTS];
+        for i in 0..N_COEFFICIENTS {
+            c_trig[i] = c[i] * theta1[i] / SECONDS_PER_JULIAN_CENTURY * theta[i].cos()
+        }
+        let c_trig: f64 = c_trig.iter().sum();
+        c1 / dt + 2.0 * c2 * t / dt.powi(2) + c_trig
+    }
+}
+
+impl RotationalElements for planets::Jupiter {
+    fn nutation_precession_coefficients() -> NutationPrecessionCoefficients {
```

Comment:
It struck me that we can all but guarantee that the compiler will inline these method implementations using `#[inline(always)].

It doesn't work on methods declarations with no bodies (i.e. where they're declared in the trait definition), but it _can_ be used on all the implementations and the associated functions of the trait that do have bodies.

Reply:
We should give it a try once we have some benchmarks in place.